src/transformers/modeling_utils.py

```diff
@@ -3054,9 +3055,14 @@ def from_pretrained(
                         if resolved_archive_file is not None:
                             is_sharded = True
                         elif use_safetensors:
-                            raise EnvironmentError(
-                                f" {_add_variant(SAFE_WEIGHTS_NAME, variant)} or {_add_variant(SAFE_WEIGHTS_INDEX_NAME, variant)} and thus cannot be loaded with `safetensors`. Please make sure that the model has been saved with `safe_serialization=True` or do not set `use_safetensors=True`."
+                            resolved_archive_file, revision = auto_conversion(
+                                pretrained_model_name_or_path, filename, **cached_file_kwargs
                             )
+                            cached_file_kwargs["revision"] = revision
+                            if resolved_archive_file is None:
+                                raise EnvironmentError(
+                                    f" {_add_variant(SAFE_WEIGHTS_NAME, variant)} or {_add_variant(SAFE_WEIGHTS_INDEX_NAME, variant)} and thus cannot be loaded with `safetensors`. Please make sure that the model has been saved with `safe_serialization=True` or do not set `use_safetensors=True`."
+                                )
```

Comment:
Putting the conversion under the `use_safetensors` flag is a great idea.

If the conversion would not be done, it would error out, so doing the conversion here is great. I'd add a bit of logging maybe.